src/core/Akka/Dispatch/ActorTaskScheduler.cs

```diff
@@ -142,6 +152,8 @@ public static void RunTask(Func<Task> asyncAction)
             ActorTaskScheduler actorScheduler = context.TaskScheduler;
             actorScheduler.CurrentMessage = context.CurrentMessage;
```

Comment:
I'm thinking, after having just looked at this code for a bunch of Phobos issues (i.e. https://github.com/petabridge/phobos-issues/issues/19) - I wonder if we should just preserve the entire `Context` here inside the default `ActorTaskScheduler`, since it seems like this issue could possibly come up inside an end-user actor and not just during unit tests.

Reply:
Well, to be completely honest, I am not sure about the reason why this was failing only in test actors so far. All that I can tell is that if I change TestActor to real actor in verification spec, it is passing the test.

So yeah, there is a chance that the issue is more serious. But this is a core part of the framework, and I am not sure how to fix it properly - that's why I found kind of "easy fix" for the test actor case. We can merge this as is to make it working better then it was before, and keep an eye on other user requests that complain to `async` support. But I do not mind dropping this PR and creating a new one with more fundamental fix... It's just I will unlikely be able to propose something better then this :)